aiohttp/server.py

```diff
@@ -291,15 +291,20 @@ def start(self):
                     reader.unset_parser()
 
                 if self._request_handler:
-                    if self._keep_alive and self._keep_alive_period:
-                        self.log_debug(
-                            'Start keep-alive timer for %s sec.',
-                            self._keep_alive_period)
-                        self._keep_alive_handle = self._loop.call_later(
-                            self._keep_alive_period, self.transport.close)
-                    elif self._keep_alive and self._keep_alive_on:
-                        # do nothing, rely on kernel or upstream server
-                        pass
+                    if self._keep_alive:
+                        sock = self.transport.get_extra_info('socket')
+                        sock.setsockopt(socket.IPPROTO_TCP,
```

Comment:
Might want to make sure a socket was actually returned and the type of the socket is TCP.